hazelcast/src/test/java/com/hazelcast/map/MapListenerTest.java

```diff
@@ -16,82 +16,141 @@
 
 package com.hazelcast.map;
 
-import com.hazelcast.config.Config;
 import com.hazelcast.core.EntryEvent;
-import com.hazelcast.core.Hazelcast;
 import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastInstanceAware;
 import com.hazelcast.core.IMap;
 import com.hazelcast.map.listener.EntryAddedListener;
 import com.hazelcast.map.listener.EntryRemovedListener;
 import com.hazelcast.map.listener.EntryUpdatedListener;
-import com.hazelcast.map.listener.MapListener;
 import com.hazelcast.query.SqlPredicate;
-import org.junit.Ignore;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import com.hazelcast.test.HazelcastTestSupport;
+import org.junit.Test;
 
 import java.io.Serializable;
-import java.util.Arrays;
 import java.util.Collection;
-import java.util.Map;
 import java.util.Random;
 import java.util.UUID;
 import java.util.concurrent.atomic.AtomicInteger;
 
-/**
- * A simple test that performs random puts, updates & removes on a map and counts the number of entries
- * and exits (with regards to the value space defined by the predicate) as performed by the map randomizer
- * and as observed on the listener side.
- */
-@Ignore("Not a JUnit test")
-public class MapListenerTest {
+import static org.junit.Assert.assertEquals;
+
+public class MapListenerTest extends HazelcastTestSupport {
 
-    private static final AtomicInteger ENTRIES, EXITS, ENTRIES_OBSERVED, EXITS_OBSERVED;
-    private static final Logger LOGGER = LoggerFactory.getLogger(MapListenerTest.class);
     private static final int AGE_THRESHOLD = 50;
 
     static {
         System.setProperty("hazelcast.map.entry.filtering.natural.event.types", "true");
-        ENTRIES = new AtomicInteger();
-        EXITS = new AtomicInteger();
-        ENTRIES_OBSERVED = new AtomicInteger();
-        EXITS_OBSERVED = new AtomicInteger();
     }
 
-    static class AllListener implements EntryAddedListener<String, Person>, EntryRemovedListener<String, Person>,
-            EntryUpdatedListener<String, Person> {
+    @Test
+    public void testListener_eventCountsCorrect() {
+        // GIVEN
+        HazelcastInstance hz = createHazelcastInstance();
 
-        private static final Logger LOGGER = LoggerFactory.getLogger(AllListener.class);
+        // GIVEN MAP & LISTENER
+        IMap<String, Person> map = hz.getMap("map");
+        AllListener listener = new AllListener();
+        map.addEntryListener(listener, new SqlPredicate("age > " + AGE_THRESHOLD), true);
+
+        // GIVEN MAP EVENTS
+        MapRandomizer mapRandomizer = new MapRandomizer(map, listener);
+        Thread t = new Thread(mapRandomizer);
+        t.start();
+        sleepAtLeastSeconds(2);
+        mapRandomizer.setRunning(false);
+
```

Comment:
`assertJoinable(t);` would be a good addition here, so we make sure the thread is gone before finishing the test.

I guess you could also use `Future future = spawn(mapRandomizer);` from `HazelcastTestSupport` and use the future for this.

Same for the other test.

Reply:
Done.